ts/src/hashkey.ts

```diff
@@ -0,0 +1,4313 @@
+            //                     "side": "BUY_OPEN",
+            //                     "timeInForce": "GTC",
+            //                     "status": "NEW",
+            //                     "priceType": "INPUT",
+            //                     "isLiquidationOrder": false,
+            //                     "indexPrice": "0",
+            //                     "liquidationType": ""
+            //                 }
+            //             },
+            //             {
+            //                 "code": "0207",
+            //                 "msg": "Create limit order sell price too low"
+            //             }
+            //         ]
+            //     }
+            //
+        } else {
+            throw new NotSupported (this.id + ' ' + 'createOrderRequest() is not supported for ' + market['type'] + ' type of markets');
+        }
+        const result = this.safeList (response, 'result', []);
+        const responseOrders = [];
+        for (let i = 0; i < result.length; i++) {
+            const responseEntry = this.safeDict (result, i, {});
+            const responseOrder = this.safeDict (responseEntry, 'order', {});
+            responseOrders.push (responseOrder);
+        }
+        return this.parseOrders (responseOrders);
+    }
+
+    async cancelOrder (id: string, symbol: Str = undefined, params = {}) {
+        /**
+         * @method
+         * @name hashkey#cancelOrder
+         * @description cancels an open order
+         * @see https://hashkeyglobal-apidoc.readme.io/reference/cancel-order
+         * @see https://hashkeyglobal-apidoc.readme.io/reference/cancel-futures-order
+         * @param {string} id order id
+         * @param {string} symbol unified symbol of the market the order was made in
+         * @param {object} [params] extra parameters specific to the exchange API endpoint
+         * @param {string} [params.type] 'spot' or 'swap' - the type of the market to fetch entry for (default 'spot')
+         * @param {string} [params.clientOrderId] a unique id for the order that can be used as an alternative for the id
+         * @param {bool} [params.trigger] *swap markets only* true for canceling a trigger order (default false)
+         * @param {bool} [params.stop] *swap markets only* an alternative for trigger param
+         * @returns {object} An [order structure]{@link https://docs.ccxt.com/#/?id=order-structure}
+         */
+        const methodName = 'cancelOrder';
+        this.checkTypeParam (methodName, params);
+        await this.loadMarkets ();
+        const request: Dict = {};
+        if (id !== undefined) {
+            request['orderId'] = id;
+        } else {
+            const clientOrderId = this.safeString (params, 'clientOrderId');
+            if (clientOrderId === undefined) {
+                throw new ArgumentsRequired (this.id + ' ' + methodName + '() requires an id argument or clientOrderId parameter');
+            }
+        }
```

Comment:
we should do the opposite here, first check if cancelOrderId is available and use it, otherwise default to id, because `id` is expected to be defined according to the signature so users can provide smth like `id="random"` and `clientOrderId` in `params`